libs/backend/src/scan/cast_vote_records/cast_vote_record_exporter.ts

```diff
@@ -0,0 +1,415 @@
+ */
+export interface CastVoteRecordExporterApi {
+  exportCastVoteRecordsToUsbDrive(
+    resultSheets: ResultSheet[] | Generator<ResultSheet>
+  ): Promise<Result<void, ExportCastVoteRecordToUsbDriveError>>;
+}
+
+/**
+ * The implementation of the cast vote record exporter API
+ */
+export class CastVoteRecordExporter implements CastVoteRecordExporterApi {
+  private readonly artifactAuthenticator: ArtifactAuthenticatorApi;
+  private readonly exporter: Exporter;
+  private readonly scannerStore: ScannerStore;
+  private readonly usbDrive: UsbDrive;
+
+  constructor(input: {
+    artifactAuthenticator: ArtifactAuthenticatorApi;
+    scannerStore: ScannerStore;
+    usbDrive: UsbDrive;
+  }) {
+    this.artifactAuthenticator = input.artifactAuthenticator;
+    this.scannerStore = input.scannerStore;
+    this.usbDrive = input.usbDrive;
+
+    this.exporter = new Exporter({
+      allowedExportPatterns: SCAN_ALLOWED_EXPORT_PATTERNS,
+      getUsbDrives: async () => {
+        const drive = await this.usbDrive.status();
+        return drive.status === 'mounted' ? [drive] : [];
+      },
+    });
+  }
+
+  async exportCastVoteRecordsToUsbDrive(
+    resultSheets: ResultSheet[] | Generator<ResultSheet>
+  ): Promise<Result<void, ExportCastVoteRecordToUsbDriveError>> {
+    const scannerState = this.getScannerState();
+
+    const exportDirectoryPathRelativeToUsbMountPoint =
+      this.getExportDirectoryPathRelativeToUsbMountPoint(scannerState);
+
+    for (const resultSheet of resultSheets) {
+      const exportCastVoteRecordFilesResult =
+        await this.exportCastVoteRecordFilesToUsbDrive(
+          scannerState,
+          resultSheet,
+          exportDirectoryPathRelativeToUsbMountPoint
+        );
+      if (exportCastVoteRecordFilesResult.isErr()) {
+        return exportCastVoteRecordFilesResult;
+      }
+    }
+
+    const exportMetadataFileResult = await this.exportMetadataFileToUsbDrive(
+      scannerState,
+      this.buildCastVoteRecordReportMetadata(scannerState),
+      exportDirectoryPathRelativeToUsbMountPoint
+    );
+    if (exportMetadataFileResult.isErr()) {
+      return exportMetadataFileResult;
+    }
+
+    return ok();
+  }
+
+  private getScannerState(): ScannerStateUnchangedByExport {
+    return {
+      batches: this.scannerStore.getBatches(),
+      definiteMarkThreshold: this.scannerStore.getDefiniteMarkThreshold(),
+      electionDefinition: assertDefined(
+        this.scannerStore.getElectionDefinition()
+      ),
+      inTestMode: this.scannerStore.getMode() === 'test',
+      pollsState: this.scannerStore.getPollsState(),
+    };
+  }
+
+  private getExportDirectoryPathRelativeToUsbMountPoint(
+    scannerState: ScannerStateUnchangedByExport
+  ): string {
+    const { electionDefinition, inTestMode } = scannerState;
+    const { election, electionHash } = electionDefinition;
+
+    let exportDirectoryName = this.scannerStore.getExportDirectoryName();
```

Comment:
One of the new pieces of state that we need to manage for continuous export is the name of the export directory. We could try to infer it, e.g. find all directories on the USB with the correct name format and then pick the one with the most recent date. But that quickly gets complicated by edge cases.